Add general style guidance based on the Release Note Improvement Initiative, Working Group 1 (#524)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/release-notes.adoc

* Do not use _future tenses_ (or "should" or "might") to describe the state after the update. You can use future tenses to refer to potential future releases.
* Use the _simple past tense_ to describe the previous situation before the current update.
* You can use the _present perfect tense_ to describe the current update. For example: 
With this update, the X component has been removed from the Y package."
* Follow the CCFR (Cause-Consequence-Fix-Result) tense logic in bug fixes.
* Do not use "now" to refer to the state after the update. For more information, see the xref:now[now] glossary entry.

[[headings-for-release-notes]]
== Headings for release notes

Introduce each release note with a heading that summarizes the release note. This practice helps customers to quickly determine if the release note is relevant to them.

* The heading can, but does not need to be, a full sentence. Do not use a period at the end of the heading.
* Use _sentence-style capitalization_, not _title case_. If necessary, headings can start with a lowercase in the case of a lowercase component name. For example: "`start` does not start the computer when `donotstop` is stopped".

> * Keep the heading under 120 characters. Write headings that are informative and specific without being overly long or too short:

Author: ```suggestion
* Write headings that are informative and specific without being overly long or too short. Adhere to the following guidelines:
** Keep the heading under 120 characters.
```

** Follow the specifics for the release notes type.
** Mention the component in a heading whenever it might not be obvious.
** Be specific; do not over-generalize headings. For example, "Program no longer crashes" is too generic.

* Do not expand abbreviations in headings. If you use an abbreviation in a heading, expand it on the first mention in the text below.

> * Do not use definitions in headings unless necessary for clarity. For example: "The `journald` system role can tune the performance of the `journald` service".

Author: ```suggestion
* Avoid definitions in headings unless necessary for clarity. For example, use definitions to disambiguate different meanings of the same name: "The `journald` system role can tune the performance of the `journald` service".
```

[[release-notes]]
= Release notes

> A release note explains specific product behavior, capability, or problems relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.

Reviewer: ```suggestion
A release note explains specific product behaviors, capabilities, or problems relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.
```

Author: I think we should stick with the singular here because one release note typically covers one behavior, one capability, or one problem (with exceptions like rebases which typically summarize more). WDYT?

Either way we definitely should fix the singular/plural mismatch that I produced :/

Author: ```suggestion
A release note explains specific product behavior, capability, or problem relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.
```

Reviewer: I just care about the mismatch so sticking to singular is fine by me.

> A release note explains specific product behavior, capability, or problems relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.

Author: fix grammatical number

Author: ```suggestion
A release note explains specific product behavior, capability, or problem relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.
```

[[style-advice-for-release-note-texts]]

> == Style advice for release note texts

Reviewer 2: I'm not a huge fan of the name of this section, or the fact that it mixes the actual paragraph text and the heading, but then heading has its own section down below.

Perhaps consider making (and naming) this section _just_ about the actual paragraph text, and leave any guidance for headings in the heading section?

You could also consider whether there is anything else that could be pulled out into their own subsections, like you did with headings and tenses. There is just a lot of bullets here, maybe they could be grouped further. Perhaps, capitalization, for example.

I personally would rather see shorter categories/sections, even if there were only 2-3 bullets, rather than longer lists of bullets.

> The normal stylistic guidelines for documentation from IBM Style and the RH SSG apply also to release note texts, particularly the following:

Author: I think referring to it as "the _IBM Style_ guide" is correct, with "guide" not being part of the name but rather a descriptor. Italicizing the references is consistent with the rest of the SSG, so I agree with it.

> The normal stylistic guidelines for documentation from the _IBM Style_ guide and the _ Red Hat supplementary style guide for product documentation_ apply also to release note texts, particularly the following:

Author: ```suggestion
The normal stylistic guidelines for documentation from the _IBM Style_ guide and the _Red Hat supplementary style guide for product documentation_ apply also to release note texts, particularly the following:
```

* When you first mention a utility, package, command, or similar item outside of a heading, define it. Do not assume that the customer is familiar with it. 

> * Omit the definition in later occurrences. If the context is ambiguous, for example, when the release note text mentions both an `example` package and an `example` service, you can repeat the definition to add clarity.

Reviewer 3: "Omit the definition" related to the previous bullet point? We have not defined what a "definition is".

> * Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.

Reviewer 3: ```suggestion
* Where possible, avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
```

Author: Not sure about the wording change; I think we've tried to simplify this in the review with Ingrid.

Incorrect bullet by me though.

```suggestion
** Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
```

Reviewer 4: I think that "Avoid" includes the sense of "where possible"--especially when you have the "but" clause that follows.

> ** Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.

Reviewer 2: This is duplicating what is said specifically in the heading section below, but that point doesn't clarify that you can if you need to for clarity reasons.

Consider clarifying it down there and then removing the duplicate here.

> * Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. See: Capitalization in _IBM Style_.

Reviewer 2: Same comment as others

```suggestion
* Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. For more information, see "Capitalization" in the _IBM Style_ guide.
```

> * Do not start a sentence with a word in lower case. You can repeat a definition to avoid starting a sentence with a lowercase name. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text[Capitalization (IBM Style)].

Reviewer 3: The RH Supp. guide does not have links to material behind a firewall, will this guideance but available to open-source community members? If so, the links to the IBM SG, won't work for them.

Reviewer 4: Good point. We could mention IBM Style without the link for those who have access.

Author: RH SSG doesn't contain links but does include refereces to IBM Style ("See the IBM Style guide for additional guidance on using IP addresses in documentation.") From my perspective, including the link is useful for those who have access and is as useful as a linkless reference to those who don't.

Reviewer 4: @bergerhoffer @mportman12 Do you see any reason to revisit the SSH's policy towards non-public links?

> * Do not start a sentence with a word in lower case. You can repeat a definition to avoid starting a sentence with a lowercase name. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text[Capitalization (IBM Style)].

Reviewer: ```suggestion
* Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text[Capitalization (IBM Style)].
```

> * Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text[Capitalization (IBM Style)].

Author: I'd say the guidance is accurate as is. Headings can start with lower case letters, sentences can't. Headings typically aren't sentences (though they may be, and in that case I'd say they should start with a capital.)

> * Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text[Capitalization (IBM Style)].

Author: ```suggestion
* Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. See: Capitalization in _IBM Style_.
```

> * Expand abbreviations in descriptions. Do not expand abbreviations in headings. See: Abbreviations in _IBM Style_.

Reviewer 2: Same as before with writing in our preferred format

```suggestion
* Expand abbreviations in descriptions. Do not expand abbreviations in headings. For more information, see "Abbreviations" in the _IBM Style_ guide.
```

> * When comparing versions, use "later" and "earlier" instead of "newer" and "older". See: Later in _IBM Style_.

Reviewer 2: ```suggestion
* When comparing versions, use "later" and "earlier" instead of "newer" and "older". For more information, see "later" and "earlier" in the _IBM Style_ guide.
```

> * Expand abbreviations in descriptions. Do not expand abbreviations in headings. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-abbreviations[Abbreviations (IBM Style)].

Author: Yes, this is intentional, as we discussed in the gdoc draft.

> * Expand abbreviations in descriptions. Do not expand abbreviations in headings. See: link:https://www.ibm.com/docs/en/ibm-style?topic=grammar-abbreviations[Abbreviations (IBM Style)].

Author: ```suggestion
* Expand abbreviations in descriptions. Do not expand abbreviations in headings. See: Abbreviations in _IBM Style_.
```

> * When comparing versions, use "later" and "earlier" instead of "newer" and "older". See: link:https://www.ibm.com/docs/en/ibm-style?topic=word-usage#later[Later (IBM Style)].

Author: ```suggestion
* When comparing versions, use "later" and "earlier" instead of "newer" and "older". See: Later in _IBM Style_.
```

* Avoid complicated syntax, such as passive voice and modal verbs, and ambiguous language. For example, replace "Should XY happen" with "If XY happens".

> * Keep admonitions to a minimum, and avoid placing multiple admonitions in a single note. Do not begin a release note with an admonition. See xref:admonitions[Admonitions (RH SSG)].

Reviewer 3: Glad to see this guidance!

> * Keep admonitions to a minimum, and avoid placing multiple admonitions in a single note. Do not begin a release note with an admonition. See xref:admonitions[Admonitions (RH SSG)].

Reviewer: ```suggestion
* Keep admonitions to a minimum, and avoid placing multiple admonitions in a single note. Do not begin a release note with an admonition. See xref:admonitions[Admonitions (Red Hat supplementary style guide)].
```

> * Avoid using telegraphic language common in merge requests and changelogs, for example, "Remove deprecated support macros".

Author: Reworded in the suggestion for the whole list.

[[tenses-in-release-notes]]
== Tenses in release notes

> Write the release notes from the perspective of just after the release, which is when most of the customers will read them. The state before the update is in the past, the state after the update is in the present, any future updates are in the future.

Reviewer 4: In the rewrite, you need a comma in this sentence. These are 2 independent clauses joined by "and". 

The state before the update is in the **past, and** the state after the update is in the present.

> Write the release notes from the perspective of just after the release, which is when most of the customers will read them. The state before the update is in the past, the state after the update is in the present, any future updates are in the future.

Author: The mention of the future was there because developers often use the future tense in their drafts, but usually to mean the state just after the update. Maybe we should rephrase it to be clearer:

`Write the release notes from the perspective of just after the release, which is when most of the customers read release notes. The state before the update is in the past and the state after the update is in the present. Do not use the future tense to refer to the current release.`

@bergerhoffer WDYT?

* Use the *simple present tense* as much as possible.
* Do not use *future tenses* (or "should" or "might") to describe the state after the update. You can use future tenses to refer to future releases.
* Use the *simple past tense* to describe the previous situation before the current update.
* You can use the *present perfect tense* to describe the current update. “With this update, the X component has been removed from the Y package.”

> * Follow the CCFR (Cause-Consequence-Fix-Result) tense logic in bug fixes.

Reviewer 3: And CCWR for known issues: 

    Cause – the user action or circumstances that trigger the bug
    Consequence – what the user experience is when the bug occurs
    Workaround - if available
    Result – what happens now that the patch is applied

Reviewer 4: Yes, it makes sense to include known issues and explain what Cause-Consequence-Fix-Result means in addition to spelling out the acronym.

Author: Darragh, I'm not sure that the CCWR has a _tense logic_ per se; all of those are typically in the present tense, so it does not make much sense to me to include it here in the Tenses section.

I'm all for including the CCWR breakdown more specifically in the Known issues section later.

* Do not use “now” to refer to the state after the update. See: xref::now[Now (RH SSG)].

[[headings-for-release-notes]]
== Headings for release notes

Introduce each release note with a heading that summarizes the release note. This practice helps customers to quickly determine if the release note is relevant to them.

* The heading can, but does not need to be, a full sentence. Do not use a period at the end of the heading.

> * Use Sentence-style capitalization (not Title Case). Headings can start with a lowercase letter (such as with a lowercase name of a component).

Reviewer 3: ```suggestion
* Use sentence-style capitalization (not Title Case). Headings can start with a lowercase letter (such as with a lowercase name of a component).
```

Author: The intention here was to showcase the capitalization of the term itself. Perhaps highlighting it would help? WDYT?

```suggestion
* Use _Sentence-style capitalization_ (not _Title Case_). Headings can start with a lowercase letter (such as with a lowercase name of a component).
```

+
----
.`start` does not start the computer when `donotstop` is stopped
----

* Keep the heading under 120 characters. Write headings that are informative and specific without being overly long or too short:

> ** Follow the specifics for the release notes type.

Author: This will be part of the follow-up PR with specific guidance for each RN type. For example, Enhancement, Known issue, Fixed issue.

> ** Follow the specifics for the release notes category.

Author: ```suggestion
** Follow the specifics for the release notes type.
```

> * Keep the heading shorter than 120 characters. Make the heading short and catchy for the reader but not too short:

Reviewer 5: ```suggestion
* Keep the heading under 120 characters. Write headings that are informative and specific without being overly long or too short:
```

Reviewer 5: Just a suggestion. Wasn't sure about using `catchy`.

Author: I like it better, thanks!

** Mention the component in a heading whenever it might not be obvious.
** Be specific; do not over-generalize headings (“Program no longer crashes” or “Program is no longer vulnerable” are over-generalized).

> * Do not expand abbreviations in headings.

Author: ```suggestion
* Do not expand abbreviations in headings. If you use an abbreviation in a heading, expand it on the first mention in the text below.
```

> * Do not use definitions in headings unless necessary for clarity.

Author: Good idea, will add an example.

```suggestion
* Do not use definitions in headings unless necessary for clarity. For example: "The `journald` system role can tune the performance of the `journald` service".
```

> * Do not use definitions in headings, unless necessary.

Reviewer: ```suggestion
* Do not use definitions in headings unless necessary for clarity.
```

[[release-notes]]
= Release notes

A release note explains specific product behavior, capability, or problem relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.

[[style-advice-for-release-note-texts]]
== Style advice for release note texts

The normal stylistic guidelines for documentation from the _IBM Style_ guide and the _Red Hat supplementary style guide for product documentation_ apply also to release note texts, particularly the following:

> 

Author: ```suggestion

* Be concise. Focus on the impact on the user, and omit any overly technical details.
```

[[release-notes]]
= Release notes

A release note explains specific product behavior, capability, or problem relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.

[[style-advice-for-release-note-texts]]
== Style advice for release note texts

The normal stylistic guidelines for documentation from the _IBM Style_ guide and the _Red Hat supplementary style guide for product documentation_ apply also to release note texts, particularly the following:

> * Be concise. Focus on the impact on the user, and omit any overly technical details.
> * When you first mention a utility, package, command, or similar item outside of a heading, define it. Do not assume that the customer is familiar with it. 
> ** Omit the definition in later occurrences. If the context is ambiguous, for example, when the release note text mentions both an `example` package and an `example` service, you can repeat the definition to add clarity.
> ** Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
> * Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. For more information, see "Capitalization" in the _IBM Style_ guide.
> * Expand abbreviations in descriptions. Do not expand abbreviations in headings. For more information, see "Abbreviations" in the _IBM Style_ guide.
> * When comparing versions, use "later" and "earlier" instead of "newer" and "older". For more information, see "later" and "earlier" in the _IBM Style_ guide.
> * Avoid complicated syntax, such as passive voice and modal verbs, and ambiguous language. For example, replace "Should XY happen" with "If XY happens".
> * Keep admonitions to a minimum, and avoid placing multiple admonitions in a single note. Do not begin a release note with an admonition. For more information, see xref:admonitions[Admonitions].
> * Avoid using telegraphic language common in merge requests and changelogs, for example, "Remove deprecated support macros".

Author: ```suggestion
Be clear and concise:
* Focus on the impact on the user, and omit any overly technical details.
* Avoid complicated syntax, such as passive voice and modal verbs, and ambiguous language. For example, replace "Should XY happen" with "If XY happens".
* Write easily readable text. Avoid using infinitive statements that are common in merge requests and changelogs, for example, "Remove deprecated support macros".

Define unfamiliar terms:
* When you first mention a utility, package, command, or similar item outside of a heading, define it. Do not assume that the customer is familiar with it.
* Omit the definition in later occurrences. If the context is ambiguous, for example, when the release note text mentions both an `example` package and an `example` service, you can repeat the definition to add clarity.
* Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
* Expand abbreviations in descriptions. Do not expand abbreviations in headings. For more information, see "Abbreviations" in the _IBM Style_ guide.

Use correct capitalization:
* Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. For more information, see "Capitalization" in the _IBM Style_ guide.

Keep admonitions to a minimum:
* Avoid placing multiple admonitions in a single note.
* Do not begin a release note with an admonition.
* For more information, see xref:admonitions[Admonitions].

```

Author: @bergerhoffer Tried reordering the advice into more shorter lists, WDYT?

* Use the _simple present tense_ as much as possible.
* Do not use _future tenses_ (or "should" or "might") to describe the state after the update. You can use future tenses to refer to potential future releases.
* Use the _simple past tense_ to describe the previous situation before the current update.
* You can use the _present perfect tense_ to describe the current update. For example: 
With this update, the X component has been removed from the Y package."
* Follow the CCFR (Cause-Consequence-Fix-Result) tense logic in bug fixes.
* Do not use "now" to refer to the state after the update. For more information, see the xref:now[now] glossary entry.

[[headings-for-release-notes]]
== Headings for release notes

Introduce each release note with a heading that summarizes the release note. This practice helps customers to quickly determine if the release note is relevant to them.

* The heading can, but does not need to be, a full sentence. Do not use a period at the end of the heading.
* Use _sentence-style capitalization_, not _title case_. If necessary, headings can start with a lowercase in the case of a lowercase component name.
+

> ----
> .`start` does not start the computer when `donotstop` is stopped

Author: ```suggestion
Example AsciiDoc:
----
.`start` does not start the computer when `donotstop` is stopped
```

> * Use _sentence-style capitalization_, not _title case_. If necessary, headings can start with a lowercase in the case of a lowercase component name.
> +
> ----
> .`start` does not start the computer when `donotstop` is stopped
> ----

Author: ```suggestion
* Use _sentence-style capitalization_, not _title case_. If necessary, headings can start with a lowercase in the case of a lowercase component name. For example: "`start` does not start the computer when `donotstop` is stopped".
```

[[release-notes]]
= Release notes

A release note explains specific product behavior, capability, or problem relevant to a product version. Release notes for a particular product version are collected in a document that is published when the new version is released.

[[style-advice-for-release-note-texts]]
== Style advice for release note texts

The normal stylistic guidelines for documentation from the _IBM Style_ guide and the _Red Hat supplementary style guide for product documentation_ apply also to release note texts, particularly the following:

> Be clear and concise:
> * Focus on the impact on the user, and omit any overly technical details.
> * Avoid complicated syntax, such as passive voice and modal verbs, and ambiguous language. For example, replace "Should XY happen" with "If XY happens".
> * Write easily readable text. Avoid using infinitive statements that are common in merge requests and changelogs, for example, "Remove deprecated support macros".
> 
> Define unfamiliar terms:
> * When you first mention a utility, package, command, or similar item outside of a heading, define it. Do not assume that the customer is familiar with it.
> * Omit the definition in later occurrences. If the context is ambiguous, for example, when the release note text mentions both an `example` package and an `example` service, you can repeat the definition to add clarity.
> * Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
> * Expand abbreviations in descriptions. Do not expand abbreviations in headings. For more information, see "Abbreviations" in the _IBM Style_ guide.
> 
> Use correct capitalization:
> * Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. For more information, see "Capitalization" in the _IBM Style_ guide.
> 
> Keep admonitions to a minimum:
> * Avoid placing multiple admonitions in a single note.
> * Do not begin a release note with an admonition.
> * For more information, see xref:admonitions[Admonitions].

Author: ```suggestion
Be clear and concise:

* Focus on the impact on the user, and omit any overly technical details.
* Avoid complicated syntax, such as passive voice and modal verbs, and ambiguous language. For example, replace "Should XY happen" with "If XY happens".
* Write easily readable text. Avoid using infinitive statements that are common in merge requests and changelogs, for example, "Remove deprecated support macros".

Define unfamiliar terms:

* When you first mention a utility, package, command, or similar item outside of a heading, define it. Do not assume that the customer is familiar with it.
* Omit the definition in later occurrences. If the context is ambiguous, for example, when the release note text mentions both an `example` package and an `example` service, you can repeat the definition to add clarity.
* Avoid definitions in headings, but you can use them to disambiguate different meanings of the same name.
* Expand abbreviations in descriptions. Do not expand abbreviations in headings. For more information, see "Abbreviations" in the _IBM Style_ guide.

Use correct capitalization:

* Do not start a sentence with a word in lowercase. You can repeat a definition to avoid starting a sentence with a lowercase name. For more information, see "Capitalization" in the _IBM Style_ guide.

Keep admonitions to a minimum:

* Avoid placing multiple admonitions in a single note.
* Do not begin a release note with an admonition.
* For more information, see xref:admonitions[Admonitions].
```

[[tenses-in-release-notes]]
== Tenses in release notes

Write the release notes from the perspective of just after the release, which is when most of the customers read release notes. The state before the update is in the past and the state after the update is in the present.

* Use the _simple present tense_ as much as possible.

> * Do not use _future tenses_ (or "should" or "might") to describe the state after the update. You can use future tenses to refer to potential future releases.

Author: ```suggestion
* Do not use _future tenses_ (or "should" or "might") to describe the state after the update.```

* Use the _simple past tense_ to describe the previous situation before the current update.
* You can use the _present perfect tense_ to describe the current update. For example: 
With this update, the X component has been removed from the Y package."
* Follow the CCFR (Cause-Consequence-Fix-Result) tense logic in bug fixes.
* Do not use "now" to refer to the state after the update. For more information, see the xref:now[now] glossary entry.

[[headings-for-release-notes]]
== Headings for release notes

Introduce each release note with a heading that summarizes the release note. This practice helps customers to quickly determine if the release note is relevant to them.

* The heading can, but does not need to be, a full sentence. Do not use a period at the end of the heading.

> * Use _sentence-style capitalization_, not _title case_. If necessary, headings can start with a lowercase in the case of a lowercase component name. For example: "`start` does not start the computer when `donotstop` is stopped".

Author: The clarifications are not recommended in headings, which this is an example of. I'll fix the rendering, thanks!
